prod/westeurope/internal/api/functions_iopayportal/function_app/terragrunt.hcl

```diff
@@ -0,0 +1,98 @@
+dependency "subnet" {
+  config_path = "../subnet"
+}
+
+# External
+dependency "app_service_pagopaproxyprod" {
+  config_path = "../../../../external/pagopaproxyprod/app_service"
+}
+
+# Internal
+dependency "resource_group" {
+  config_path = "../../../resource_group"
+}
+
+dependency "subnet_apimapi" {
+  config_path = "../../../api/apim/subnet"
+}
+
+# Common
+dependency "virtual_network" {
+  config_path = "../../../../common/virtual_network"
+}
+
+dependency "application_insights" {
+  config_path = "../../../../common/application_insights"
+}
+
+dependency "key_vault" {
+  config_path = "../../../../common/key_vault"
+}
+
+# Include all settings from the root terragrunt.hcl file
+include {
+  path = find_in_parent_folders()
+}
+
+terraform {
+  source = "git::git@github.com:pagopa/io-infrastructure-modules-new.git//azurerm_function_app?ref=v2.1.10"
+}
+
+inputs = {
+  name                = "iopayportal"
+  resource_group_name = dependency.resource_group.outputs.resource_name
+
+  app_service_plan_info = {
+    kind     = "elastic"
+    sku_tier = "ElasticPremium"
+    sku_size = "EP1"
+  }
+
+  runtime_version = "~3"
+
+  pre_warmed_instance_count = 2
+
+  application_insights_instrumentation_key = dependency.application_insights.outputs.instrumentation_key
+
+  app_settings = {
+    FUNCTIONS_WORKER_RUNTIME       = "node"
+    WEBSITE_NODE_DEFAULT_VERSION   = "12.18.0"
+    WEBSITE_RUN_FROM_PACKAGE       = "1"
+    FUNCTIONS_WORKER_PROCESS_COUNT = 4
+    NODE_ENV                       = "production"
+
+    # DNS configuration to use private dns zones
+    // TODO: Use private dns zone https://www.pivotaltracker.com/story/show/173102678
+    //WEBSITE_DNS_SERVER     = "168.63.129.16"
+    //WEBSITE_VNET_ROUTE_ALL = 1
+
+    
+
+    SERVICES_API_URL = "http://api-internal.io.italia.it/"
+
+    // PAGOPA : endpoints use to call pagopa-proxy service : getPaymentInfo, activatePayment, getActivationStatus
+    IO_PAGOPA_PROXY_API_TOKEN : ""
+    IO_PAGOPA_PROXY_PROD_BASE_URL = "https://${dependency.app_service_pagopaproxyprod.outputs.default_site_hostname}"
+    IO_PAGOPA_PROXY_TEST_BASE_URL = "https://${dependency.app_service_pagopaproxytest.outputs.default_site_hostname}"
+    PAGOPA_BASE_PATH    = "/pagopa/api/v1"
```

Comment:
on staging slot these parameters are missing.
if IO_PAGOPA_PROXY_API_TOKEN it's a secret you need to add this section (also remember to add the secret into keyvault)
```
  app_settings_secrets = {
    key_vault_id = dependency.key_vault.outputs.id
    map = {
      IO_PAGOPA_PROXY_API_TOKEN = "iopayportal-IO-PAGOPA-PROXY-API-TOKEN"
    }
  }
```

Reply:
if function doesn't require any secret, add (also on function_slot)
```
  app_settings_secrets = {
    key_vault_id = "dummy"
    map          = {}
  }
```

Reply:
currently we do not know, probably not, so we set `app_settings_secrets` to `dummy` tnx